Spigot-Server-Patches/0460-Brand-support.patch

```diff
@@ -0,0 +1,81 @@
+From 899faf727c64d5a089a6c900a0ae15b0bd14b9b5 Mon Sep 17 00:00:00 2001
+From: DigitalRegent <misterwener@gmail.com>
+Date: Mon, 6 Apr 2020 21:21:23 +0200
+Subject: [PATCH] Brand support
+
+
+diff --git a/src/main/java/net/minecraft/server/PlayerConnection.java b/src/main/java/net/minecraft/server/PlayerConnection.java
+index 7929fcc8..e224be23 100644
+--- a/src/main/java/net/minecraft/server/PlayerConnection.java
++++ b/src/main/java/net/minecraft/server/PlayerConnection.java
+@@ -106,6 +106,8 @@ public class PlayerConnection implements PacketListenerPlayIn {
+     private static final int MAX_SIGN_LINE_LENGTH = Integer.getInteger("Paper.maxSignLength", 80);
+     private static final long KEEPALIVE_LIMIT = Long.getLong("paper.playerconnection.keepalive", 30) * 1000; // Paper - provide property to set keepalive limit
+ 
++    private String clientBrandName = null; // Paper - Brand name
++
+     public PlayerConnection(MinecraftServer minecraftserver, NetworkManager networkmanager, EntityPlayer entityplayer) {
+         this.minecraftServer = minecraftserver;
+         this.networkManager = networkmanager;
+@@ -2641,6 +2643,8 @@ public class PlayerConnection implements PacketListenerPlayIn {
+     private static final MinecraftKey CUSTOM_REGISTER = new MinecraftKey("register");
+     private static final MinecraftKey CUSTOM_UNREGISTER = new MinecraftKey("unregister");
+ 
++    private static final MinecraftKey MINECRAFT_BRAND = new MinecraftKey("brand"); // Paper - Brand support
++
+     @Override
+     public void a(PacketPlayInCustomPayload packetplayincustompayload) {
+         PlayerConnectionUtils.ensureMainThread(packetplayincustompayload, this, this.player.getWorldServer());
+@@ -2665,10 +2669,16 @@ public class PlayerConnection implements PacketListenerPlayIn {
+                 this.disconnect("Invalid payload UNREGISTER!");
+             }
+         } else {
++            // Paper start - Brand support
```

Comment:
any reason you moved this out of the try block?
I would just have added the if as a one liner in the try block, minimizes diff and doesnt change exception behavior.

Reply:
Nothing can ever go wrong there. Maybe just `IndexOutOfBoundsException ` for readBytes(data)` but that's is impossible since length of array `data` is never greater than length of readable bytes.

Reply:
Generally, min diff is priority. I would've left this in the try block.